ee/tabby-webserver/src/service/answer.rs

```diff
@@ -490,6 +505,119 @@ Remember, don't blindly repeat the contexts verbatim. When possible, give code s
     )
 }
 
+use anyhow::Result;
+
+static GLOBAL_REPOS: OnceCell<HashMap<String, Repository>> = OnceCell::const_new();
+
+/// Combine code snippets from search results rather than utilizing multiple hits: Presently, there is only one rule: if the number of lines of code (LoC) is less than 200, and there are multiple hits (number of hits > 1), include the entire file.
+pub async fn merge_code_snippets(
+    repository: Arc<dyn RepositoryService>,
+    hits: Vec<CodeSearchHit>,
+    policy: &AccessPolicy,
+    source_id: &str,
+) -> Vec<CodeSearchHit> {
+    // group hits by filepath
+    let mut file_hits: HashMap<String, Vec<CodeSearchHit>> = HashMap::new();
+    for hit in hits {
+        let key = format!("{}-{}", source_id, hit.doc.filepath);
+        file_hits.entry(key).or_default().push(hit);
+    }
+
+    let repos = GLOBAL_REPOS
```

Comment:
repos shouall be compute on each call to `merge_code_snippets`. Shouldn't create a global variable here.